PatchesAndAreasApi/V1/Controllers/PatchesAndAreasApiController.cs

```diff
@@ -138,6 +144,33 @@ public async Task<IActionResult> UpdatePatchForResponsibility([FromRoute] Update
 
         }
 
+        [HttpPatch]
+        [Route("{id}/responsibleEntities")]
+        [LogCall(LogLevel.Information)]
+        public async Task<IActionResult> ReplacePatchResponsibleEntities([FromRoute] PatchesQueryObject query,
+                                                                          [FromBody] List<ResponsibleEntities> responsibleEntitiesRequestObject)
+        {
+            var contextHeaders = _contextWrapper.GetContextRequestHeaders(HttpContext);
+            var ifMatch = GetIfMatchFromHeader();
+
+            try
+            {
+                // We use a request object AND the raw request body text because the incoming request will only contain the fields that changed
+                // whereas the request object has all possible updateable fields defined.
```

Comment:
is this all necessary if we're doing a patch.ResponsibleEntities = responsibleEntitiesRequestObject; in the gateway? we're not doing a field by field comparison....

Reply:
The reason why we are not doing a field-by-field comparison is because we want this endpoint to meet the following use cases:

- Be able to switch housing officers 
- Be able to add new housing officers (existing or none existing)
- be able to have more than one housing officer assigned to a patch
if we do a field-by-field comparison then we would not be able to add new housing officers or have more than one housing officer assigned to a patch

I do agree with @LBHCallumM point that this endpoint should be a PUT rather than PATCH. 
Happy to change that if all agree.

Reply:
But then why have all the above?

Reply:
For the next PR as well as doing the SNS event stuff I'm going to see if I can get the endpoint to update field by field.